rimage/lazy_encoded.go

```diff
@@ -49,25 +53,40 @@ func NewLazyEncodedImage(imgBytes []byte, mimeType string) image.Image {
 	}
 }
 
-func (lei *LazyEncodedImage) decode() {
+// Helper method for checking lei errors.
+func (lei *LazyEncodedImage) checkError(err interface{}) error {
+	if err != nil {
+		switch e := err.(type) {
+		case error:
+			return e
+		default:
+			return fmt.Errorf("%v", err)
+		}
+	}
+	return nil
+}
+
+// DecodeImage decodes the image. Returns nil if no errors occurred.
+// This method is idempotent.
+func (lei *LazyEncodedImage) DecodeImage() error {
 	lei.decodeOnce.Do(func() {
 		defer func() {
 			if err := recover(); err != nil {
-				lei.decodeErr = err
+				lei.decodeImageErr = err
 			}
 		}()
-		lei.decodedImage, lei.decodeErr = DecodeImage(
+		lei.decodedImage, lei.decodeImageErr = DecodeImage(
 			context.Background(),
 			lei.imgBytes,
 			lei.mimeType,
 		)
 	})
-	if lei.decodeErr != nil {
-		panic(lei.decodeErr)
-	}
+	return lei.checkError(lei.decodeImageErr)
 }
 
-func (lei *LazyEncodedImage) decodeConfig() {
+// DecodeConfig decodes the image configuration. Returns nil if no errors occurred.
+// This method is idempotent.
+func (lei *LazyEncodedImage) DecodeConfig() error {
```

Comment:
when would a caller want to decode the config but not the image or vice versa? Wouldn't you always want to decode both or neither?

Reply:
I think the stream server is a good example. IIUC, we don't actually "do anything" with the image e.g. transform it, crop it etc.

I checked for usages of `DecodedImage` or `ConvertImage` (which calls `DecodedImage` if its a lazy encoded image) and there are none for handling camera streams.

We basically check the bounds of the image (which only requires the decoded config not image contents) and then extract the raw bytes [here](https://github.com/viamrobotics/rdk/blob/8d7ef31e5192ae667e84c244daa2c9b19159c1a6/gostream/stream.go#L270-L271).